Code/Framework/AzCore/Platform/Common/UnixLike/AzCore/Debug/Trace_UnixLike.cpp

```diff
@@ -6,74 +6,123 @@
-        if (tracerPidOffset == AZStd::string_view::npos)
+        bool AttachDebugger()
         {
+            // Not supported yet
+            AZ_Assert(false, "AttachDebugger() is not supported for Unix platform yet");
             return false;
         }
-        for (size_t i = tracerPidOffset + tracerPidString.length(); i < numRead; ++i)
+
+        void SignalHandler(int handler)
+        {
+        }
+
+        void HandleExceptions(bool isEnabled)
         {
-            if (!::isspace(processStatusView[i]))
+            if (isEnabled)
+            {
+                signal(SIGSEGV, ExceptionHandler);
+                signal(SIGTRAP, ExceptionHandler);
+                signal(SIGILL, ExceptionHandler);
+            }
+            else
             {
-                return processStatusView[i] != '0';
+                signal(SIGSEGV, SIG_DFL);
+                signal(SIGTRAP, SIG_DFL);
+                signal(SIGILL, SIG_DFL);
             }
         }
-        return false;
-    }
 
-    bool IsDebuggerPresent()
-    {
-        static bool s_detectionPerformed = false;
-        static bool s_debuggerDetected = false;
-        if (!s_detectionPerformed)
+        void DebugBreak()
         {
-            s_debuggerDetected = performDebuggerDetection();
-            s_detectionPerformed = true;
+            raise(SIGINT);
         }
-        return s_debuggerDetected;
-    }
-
-    bool AttachDebugger()
-    {
-        // Not supported yet
-        AZ_Assert(false, "AttachDebugger() is not supported for Unix platform yet");
-        return false;
-    }
+#endif // AZ_ENABLE_DEBUG_TOOLS
 
-    void HandleExceptions(bool)
-    {}
+        void Terminate(int exitCode)
+        {
+            _exit(exitCode);
+        }
+    } // namespace Platform
 
-    void DebugBreak()
+#if defined(AZ_ENABLE_DEBUG_TOOLS)
+    void ExceptionHandler(int signal)
     {
-        raise(SIGINT);
-    }
-#endif // AZ_ENABLE_DEBUG_TOOLS
+        char message[MaxMessageLength];
+        Debug::Trace::Instance().Output(nullptr, "==================================================================\n");
+        azsnprintf(message, MaxMessageLength, "Error: signal %s: \n", strsignal(signal));
+        Debug::Trace::Instance().Output(nullptr, message);
 
-    void Terminate(int exitCode)
-    {
-        _exit(exitCode);
+        StackFrame frames[MaxStackLines];
+        SymbolStorage::StackLine stackLines[MaxStackLines];
+        SymbolStorage decoder;
+        const unsigned int numberOfFrames = StackRecorder::Record(frames, MaxStackLines);
+        decoder.DecodeFrames(frames,  numberOfFrames, stackLines);
+        for(int i = 0; i < numberOfFrames; ++i) {
+            azsnprintf(message, MaxMessageLength, "%s \n", stackLines[i]);
+            Debug::Trace::Instance().Output(nullptr, message);
+        }
+        Debug::Trace::Instance().Output(nullptr, "==================================================================\n");
```

Comment:
We may want to use `Trace::RawOutput` instead of `Trace::Output` as that avoids any EBus calls and therefore any mutex locks and heap allocations that might occur.